### PR TITLE
chore: reduce the number of (verification) RPC calls

### PR DESCRIPTION
**Pull request by caybro**
- when calling `Utils.getColorHashAsJson()` we don't actually need to issue ID verification requests at all and there might be situations when we know beforehand that we don't need the ENS verification either
- change these helper functions' syntax and do early returns in that case
- in MessageContextMenuView, the "contact details" were duplicated, so remove one
- remove dead code, fix some warnings

### What does the PR do

Reduces the number of verification related RPC calls

### Affected areas

A lot (mostly messaging)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Community, chat, user list, user profile:
image

Addressbook:
image

### CODE REVIEW

File: ui/imports/shared/views/chat/MessageView.qml [hunk 1]
```diff
@@ -703,6 +703,7 @@ Loader {
                     }
                 }
 
+                hasLinks: linkUrlsModel.count
```
Comment (MishkaRogachev) [inline ui/imports/shared/views/chat/MessageView.qml:706]: linkUrlsModel.count > 0

Comment (caybro) [reply ui/imports/shared/views/chat/MessageView.qml:706]: The same thing effectively, as they both evaluate to a truthy value

**Review by alaibe — APPROVED** [review]

**Review by jrainville — APPROVED** [review]
LGTM

**Comment by caybro** [comment]
Need to rebase this...

**Pull request merged**